lib/src/model/path_utils.dart

```diff
@@ -0,0 +1,36 @@
+import 'dart:ui';
+
+class RingNumberProvider<T> {
```

Comment:
This class doesn't have much to do with paths. Let's move it out of `path_utils.dart`

Reply:
Do you have any suggestions where to? 
Do you have a file for 'general utils' or similar?

Reply:
how about `ring_number_provider.dart`?